### DISCUSSION TITLE
impedance arg has no effect of edb.siwave

### DISCUSSION THREAD

**Discussion by raindigoa** [ansys/pyaedt#2111] [General]
As title,

recently I tried to set circuit ports by using the command "EdbSiwave.create_circuit_port_on_pin". 

After saving edb and closing it, I would open siwave and import the modified edb file to check the result. And it showed that the ports were set to the assigned pins successfully while the impedance keeps always 50ohm instead of any value that I gave in pyaedt.

This problem has existed since the very early version, please fix it.

Thank you.

**Comment by ghost** [comment]
I have not used this method; however, there does not appear to be a similar impedance/resistance attrubute in EdbSiwave.create_v/i_source_on_pin

**Comment by hui-zhou-a** [comment]
@raindigoa fixed.
https://github.com/pyansys/pyaedt/pull/2178

**Reply by raindigoa** [reply]
Hi ring,

thank you, the issue has been solved
